qiskit/extensions/hamiltonian_gate.py

```diff
@@ -116,6 +117,17 @@ def _define(self):
         qc._append(UnitaryGate(self.to_matrix()), q[:], [])
         self.definition = qc
 
+    def has_qasm2(self):
+        """Return `False` because no QASM2 representation is defined for HamiltonianGate."""
+        return False
+
```

Comment:
We shouldn't add more magic interfaces, and in fact this gate shouldn't be doing anything like this.  If you just remove this handling, I think the exporter will actually succeed in outputting the gate.

Reply:
I don't understand why it's magic. But if we don't need it, so much the better.

Reply:
Removed in 68c1832.